src/typecheck/core.ml

```diff
@@ -29,17 +34,54 @@ module Ae = struct
       | Type.Builtin Ast.False ->
         `Term (Base.app0 (module Type) env s T._false)
 
-      (* Terms connectors *)
+      (* Boolean operators *)
+      | Type.Builtin Ast.Not ->
+        `Term (Base.term_app1 (module Type) env s T.neg)
+      | Type.Builtin Ast.And ->
+        `Term (Base.term_app_left (module Type) env s mk_and)
+      | Type.Builtin Ast.Or ->
+        `Term (Base.term_app_left (module Type) env s mk_or)
+      | Type.Builtin Ast.Xor ->
+        `Term (Base.term_app_left (module Type) env s T.xor)
+      | Type.Builtin Ast.Imply ->
+        `Term (Base.term_app_right (module Type) env s T.imply)
+      | Type.Builtin Ast.Equiv ->
+        `Term (Base.term_app_right (module Type) env s T.equiv)
+
+      (* If-then-else *)
+      | Type.Builtin Ast.Ite ->
+        `Term (
+          Base.make_op3 (module Type) env s (fun _ (a, b, c) ->
+              let cond = Type.parse_prop env a in
+              let then_ = Type.parse_term env b in
+              let else_ = Type.parse_term env c in
+              T.ite cond then_ else_
+          )
+        )
+
+      (* Equality *)
       | Type.Builtin Ast.Eq ->
         `Term (Base.term_app2 (module Type) env s T.eq)
+      | Type.Builtin Ast.Distinct ->
+        `Term (
+          fun _ast args ->
+            let args = List.map (Type.parse_term env) args in
+            match args with
+            | [x; y] -> T.neq x y
+            | _ -> T.distinct args
+        )
+
+      (* Ignore the AC symbol *)
+      | Type.Id id when Id.equal id Id.ac_symbol ->
+        `Tags (fun _ast _args -> [])
```

Comment:
You should not ignore the `ac` specification here, but instead add a new tag for it. See e.g. how it is done for `Zf` (cf https://github.com/Gbury/dolmen/blob/f1f05bb281826fa7c808d86fbb21c514d999edad/src/interface/tag.ml#L42-L49 and https://github.com/Gbury/dolmen/blob/f1f05bb281826fa7c808d86fbb21c514d999edad/src/typecheck/core.ml#L394-L395 )

Reply:
Done.